### ISSUE TITLE
Rating component is not accessible

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-frontend#6679]
According to the WAVE tool (extension available at: https://addons.mozilla.org/en-US/firefox/addon/wave-accessibility-tool/), the `Rating` component is not accessible, i.e. it is not clear what each star represents.

Here is a sketch of patch to make it better:

```diff
diff --git a/src/ui/components/Rating/index.js b/src/ui/components/Rating/index.js
index 24e584192..e6364a81c 100644
--- a/src/ui/components/Rating/index.js
+++ b/src/ui/components/Rating/index.js
@@ -131,6 +131,9 @@ export class RatingBase extends React.Component<InternalProps, StateType> {
       if (hoveringOverStar !== null) {
         isSelected = thisRating <= hoveringOverStar;
       }
+
+      const title = this.renderTitle(rating, readOnly, thisRating);
+
       const props = {
         className: makeClassName('Rating-star', `Rating-rating-${thisRating}`, {
           'Rating-selected-star': isSelected,
@@ -140,7 +143,7 @@ export class RatingBase extends React.Component<InternalProps, StateType> {
         key: `rating-${thisRating}`,
         onClick: undefined,
         onMouseEnter: () => this.onHoverStar(thisRating),
-        title: this.renderTitle(rating, readOnly, thisRating),
+        title,
       };
 
       if (readOnly) {
@@ -151,9 +154,21 @@ export class RatingBase extends React.Component<InternalProps, StateType> {
         props.onClick = this.onSelectRating;
       }
 
+      const id = `Rating-rating-${thisRating}-title`;
+
       return (
         // eslint-disable-next-line react/jsx-key
-        <button type="button" value={thisRating} {...props} />
+        <React.Fragment>
+          <button
+            aria-describedby={id}
+            type="button"
+            value={thisRating}
+            {...props}
+          />
+          <span id={id} className="visually-hidden">
+            {title}
+          </span>
+        </React.Fragment>
       );
     });
   }
```

I believe this would be a very nice improvement.

**Comment by willdurand** [comment]
> Hi @willdurand I would like to work on it , Can I?

sure!

**Comment by AlexandraMoga** [comment]
@willdurand  I've used the [WAVE](https://addons.mozilla.org/en-US/firefox/addon/wave-accessibility-tool/) extension to try to understand what was the issue and how the solution on -dev looks like.

The ratings card inspected with WAVE looks like this:
1. For stage (before fix) - rating star function is not readable
![image](https://user-images.githubusercontent.com/31961530/48786978-96234480-ecf0-11e8-9125-d446e6e36e46.png)

2. On -dev (after fix) - rating stars functionality is readable 
![2018-11-20_1807](https://user-images.githubusercontent.com/31961530/48787233-121d8c80-ecf1-11e8-9371-ad6ff3c9b35e.png)

Questions:
1. Is this the expected outcome?
2. The 1st and the 5th stars are out of bounds:

![image](https://user-images.githubusercontent.com/31961530/48787781-2615be00-ecf2-11e8-9e34-e81b3cf87f46.png)

**Comment by willdurand** [comment]
> Is this the expected outcome?

Yes, that's what we want.

> The 1st and the 5th stars are out of bounds:

Could be the tool that modifies the page. WAVE is a testing tool, not a reader for people with disabilities.

**Comment by AlexandraMoga** [comment]
Yes, the tool modifies other page contents as well, not only the ratings section.

In that case, I'm marking the issue as verified fixed on -dev.
